lib/al/Library/Movement/EnemyStateBlowDown.cpp

```diff
@@ -0,0 +1,98 @@
+#include <al/Library/LiveActor/ActorActionFunction.h>
+#include <al/Library/LiveActor/ActorClippingFunction.h>
+#include <al/Library/LiveActor/ActorCollisionFunction.h>
+#include <al/Library/LiveActor/ActorFlagFunction.h>
+#include <al/Library/LiveActor/ActorMovementFunction.h>
+#include <al/Library/LiveActor/ActorPoseKeeper.h>
+#include <al/Library/LiveActor/ActorSensorFunction.h>
+#include <al/Library/Math/MathAngleUtil.h>
+#include <al/Library/Math/MathLengthUtil.h>
+#include <al/Library/Movement/EnemyStateBlowDown.h>
+
+namespace al {
+void EnemyStateBlowDown::start(const al::HitSensor* sensor) {
+    sead::Vector3f dir = al::getSensorPos(sensor) - al::getTrans(mActor);
+
+    al::verticalizeVec(&dir, al::getGravity(mActor), dir);
+
+    if (!al::tryNormalizeOrZero(&dir))
+        al::calcFrontDir(&dir, mActor);
+
+    start(-dir);
+}
+
+void EnemyStateBlowDown::start(const sead::Vector3f& dir) {
+    if (mParam->mFaceAwayFromActor) {
+        al::faceToDirection(mActor, -dir);
+    }
+
+    auto actor = mActor;
+    auto direction = dir * mParam->mGravityStrength;
+    auto gravity = al::getGravity(actor);
+    auto velocity = gravity * mParam->mVelocityStrength;
```

Comment:
Either `auto*` or `auto&`, but I would prefer the real type for all of them either way.